src/backend/distributed/utils/node_metadata.c

```diff
@@ -326,6 +325,52 @@ ReadWorkerNodes()
 
 
 /*
+ * RemoveNodeFromCluster removes the provided node from the pg_dist_node table of
+ * the master node and all nodes with metadata.
+ * The call to the master_remove_node should be done by the super user. If there are
+ * active shard placements on the node; the function removes the node when forceRemove
+ * flag is set, it errors out otherwise.
+ */
+static void
+RemoveNodeFromCluster(char *nodeName, int32 nodePort, bool forceRemove)
+{
+	char *nodeDeleteCommand = NULL;
+	bool hasShardPlacements = false;
+	WorkerNode *workerNode = NULL;
+
+	EnsureSchemaNode();
+	EnsureSuperUser();
+
+	hasShardPlacements = NodeHasActiveShardPlacements(nodeName, nodePort);
+	if (hasShardPlacements && forceRemove)
```

Comment:
Same comment as #1092 here, I think it would be better to include an internal `if` statement rather than repeat part of a boolean expression in a subsequent `else if`, which looks weird.